### ISSUE TITLE
Forecast RC-to-RC reconciliation

### ISSUE THREAD

**Issue by getorymckeag** [trolie/spec#146]
Design, perhaps sequence diagram.

**Comment by catkins-miso** [comment]
![image](https://github.com/user-attachments/assets/c032a78c-04d9-4498-9990-b8ad9e6c9ffe)

**Comment by catkins-miso** [comment]
One thing I should call out that's not in the picture: the calls are symmetric, both sides true-up on their own and alarm on their own. A separate supervisor could be implemented (separately) that pulls both sides' diff monitoring set and makes noise in case the TROLIE implementation is busted.

**Comment by catkins-miso** [comment]
Worth noting that there is no "seam" Monitoring Set defined here. The RC-A and RC-B monitoring sets might contain tier 1 facilities, and even the AB Monitoring Set--the intersection of RC-A and RC-B--might not contain every facility on the seam if, for example, the TO in A delegates their Ratings Obligation to the TOP who is in B.

Note also that AB Monitoring Set and AB Limits Diff Monitoring Set are dynamically defined.

**Comment by getorymckeag** [comment]
This is good, and I think might be worth a brief talk through today.  First, maybe just in this chat, can you remind me of exactly what you mean by "tier 1" facilities.

**Comment by catkins-miso** [comment]
> can you remind me of exactly what you mean by "tier 1" facilities.

It's like a monitoring exclave: Facility(ies) in your neighbor's footprint that you keep an eye on. By contrast a seam facility (an intertie) is partially in your own footprint and partially in your neighbor's.

**Comment by getorymckeag** [comment]
@catkins-miso One comment here as I'm thinking through this- the new limits diff media type: is it useful, perhaps even needed for monitoring the system?  Sure, and I think it's a good idea.  But is it needed for interop between vendors?  If not, should it still be part of TROLIE?

**Comment by catkins-miso** [comment]
> the new limits diff media type: is it needed for interop between vendors? If not, should it still be part of TROLIE?

tl;dr You're right. It doesn't belong in TROLIE.

I know we are on the same page here, but for posterity: TROLIE promotes interop between TO/TOP/RC/TP/etc. by promulgating a community-driven standard that vendors should adopt. I think of the vendor-interop within a company boundary as natural consequence of that purpose, though a happy one.

Right, so as a concrete example, I believe you're essentially asking if RC-A should be able to pull a diff from RC-B's TROLIE for any reason? It's a *really* good, pragmatic question--YAGNI.

The interop is at the process level yet supported by TROLIE implementations. Being able to reconcile nominally equivalent monitoring sets and observe their differences is key to a number coordination and support use cases. So, while not strictly necessary to exchange of any particular kind of rating, it supports successful ratings exchange. Let me try to support that claim. 

I had a couple of use cases in mind:
* Drift troubleshooting of limits snapshots (RT and forecast)---does an App Support person at RC-A want to be able to dig into what the TROLIE at RC-B thinks the AB forecast diff is when their own TROLIE implementation gives them a heads-up (via some out of band monitor on the local AB diff)?
* Model reconciliation between peers--could RC-1's lower environment TROLIE point at RC-2's production TROLIE to determine what the A monitoring set's *model* diff is? I'm thinking about safe model promotion. This is mostly about a diff in the names...

Well, I just spent 30 minutes noodling, and I haven't convinced myself. I think you're right. Until we identify a concrete use case where exchanging diffs between companies is necessary, I don't think it belongs in TROLIE.

**Comment by catkins-miso** [comment]
@getorymckeag We still have to detail exactly how a Regional Limits Snapshot is requested, but it uses the same media types and query parameters as the Global Limits Snapshot, so I was thinking a query parameter was most appropriate: `GET /limits/forecast-snapshot?type=regional`
